lib/memo_wise.rb

```diff
@@ -530,15 +568,22 @@ def reset_memo_wise(method_name = nil, *args, **kwargs)
     when MemoWise::InternalAPI::ONE_REQUIRED_KEYWORD then method_hash&.delete(kwargs.first.last)
     when MemoWise::InternalAPI::SPLAT then method_hash&.delete(args)
     when MemoWise::InternalAPI::DOUBLE_SPLAT then method_hash&.delete(kwargs)
-    else # MemoWise::InternalAPI::MULTIPLE_REQUIRED, MemoWise::InternalAPI::SPLAT_AND_DOUBLE_SPLAT
-      key = if method_arguments == MemoWise::InternalAPI::SPLAT_AND_DOUBLE_SPLAT
-              [args, kwargs]
-            else
-              method.parameters.map.with_index do |(type, name), i|
-                type == :req ? args[i] : kwargs[name]
-              end
-            end
-      method_hash&.delete(key)
+    when MemoWise::InternalAPI::SPLAT_AND_DOUBLE_SPLAT
+      method_hash&.[](args)&.delete(kwargs)
```

Comment:
I'm confused why the key for `SPLAT_AND_DOUBLE_SPLAT` doesn't continue to be `[args, kwargs]`? Is it because the `args` is handled already somehow?

Reply:
The premise of this PR is that for any values, it is faster to do:

```ruby
hash[val1][val2]
```

than it is to do:

```ruby
hash[[val1, val2]]
```

because of both simpler/faster `.hash` methods, and avoiding the instantiation of a temporary array. This premise is true even when `val1` and `val2` are `args` and `kwargs`. (This is why we see benchmark speedups for the `(a, *args, b:, **kwargs)` case.)

Does that make sense?

Reply:
Yes! Thank you!

But could we have a comment on that case, saying “first time we lookup with `args`, next time through we’ll be nested on level down and end up at (which branch so we end up at to do the `kwargs`?)

Reply:
Just added some clarifying comments. Let me know if it's still confusing.

Reply:
Thank you!